### ISSUE TITLE
Authenticate from in-memory keypair

### ISSUE THREAD

**Issue by jonhoo** [alexcrichton/ssh2-rs#77]
libssh2 has [`libssh2_userauth_publickey_frommemory`](https://www.libssh2.org/libssh2_userauth_publickey_frommemory.html), which allows you to authenticate a user using an in-memory keypair. It apparently only works when linking against OpenSSL, but given that we do that on `cfg(unix)`, providing this method under `#[cfg(unix)]` seems like something we could do?

**Comment by alexcrichton** [comment]
Sounds plausible to me!

**Comment by promethe42** [comment]
Hi there,

I want to use `libgit2`'s `git_cred_ssh_key_memory_new`. See this issue: https://github.com/alexcrichton/git2-rs/issues/320

I'm using the `ssh2` crate in my app (in addition to `git2`). Yet I see this error:

```
this version of libgit2 was not built with ssh memory credentials
```

**Comment by alexcrichton** [comment]
@promethe42 it may mean that libgit2 didn't correctly detect libssh2 support or otherwise something different is missing? I'm not entirely sure :(

**Comment by promethe42** [comment]
LibSSH2 has support for WinCNG.
When I remove the `#[cfg(unix)]` above `userauth_pubkey_memory` it builds fine.

I'll see if it also runs fine :)

**Comment by promethe42** [comment]
Ok I tried to make it work on Windows but to no avail.

What I did instead to make SSH auth more seamless is to parse `~/.ssh/config`. Grammar is here:

https://github.com/aerys/gpm/blob/master/src/gpm/ssh_config.pest

**Comment by wez** [comment]
Can this issue be closed?  It sounds like #80 resolved the original request.  If there's something else left to do, can that be opened as a separate issue?

**Comment by jonhoo** [comment]
Yes, I think you're right that this is closed by #80!
